closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -362,6 +383,45 @@ int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
         return (len + tagLenBytes);
     }
 
+    /**
+     * Performs encryption operation for the last time.
+     *
+     * @param src the input buffer with the data to be encrypted
+     * @param dst the offset in <code>in</code>
+     * @return the number of bytes placed into the <code>out</code> buffer
+     */
+    int encryptFinal(ByteBuffer src, ByteBuffer dst)
+        throws IllegalBlockSizeException, ShortBufferException {
+        byte[] dst_arr;
+
+        // Get array from source
+        byte[] src_arr = getBbArray(src);
+        int src_ofs = src.position() + (src.hasArray() ? src.arrayOffset() : 0);
+
+        // Get array from destination
+        if (dst.hasArray()) {
+            byte[] dst_arr = dst.array();
+        } else {
+            byte[] dst_arr = new byte[dst.capacity()];
+        }
```

Comment:
This won't compile: `dst_arr` will not be visible to line 409.

Reply:
Right, fixed.